libeos-updater-util/flatpak.c

```diff
@@ -1199,31 +1150,22 @@ eos_updater_util_flatpak_ref_action_application_progress_in_state_path (GCancell
     }
 
   /* Enumerate each section. The section name is the path to the file */
-  groups_iter = groups = g_key_file_get_groups (state_key_file, NULL);
+  groups = g_key_file_get_groups (state_key_file, NULL);
 
-  while (*groups_iter != NULL)
+  for (groups_iter = groups; *groups_iter != NULL; ++groups_iter)
     {
       const gchar *source_path = *groups_iter;
-      g_autofree gchar *progress_string = NULL;
-      guint64 progress;
-
-      /* We need to use g_key_file_get_value here to guard against errors */
-      progress_string = g_key_file_get_value (state_key_file,
+      gint progress = g_key_file_get_integer (state_key_file,
                                               *groups_iter,
                                               "Progress",
                                               error);
 
-      if (progress_string == NULL)
-        return NULL;
-
-      if (!parse_monotonic_counter (progress_string, &progress, error))
+      if (progress == 0)
```

Comment:
I think you need to check the `error` here. At the moment it’s possible for `Progress=0` to be set in the key file, and this code will return `NULL` with no error set.

Reply:
One of the invariants set earlier was that `Progress > 0`. That said, it wouldn't hurt to check.